### PR TITLE
Add validation transfer API for Microsoft.Billing.

### PR DESCRIPTION
**Pull request by jbnpaul**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 1]
```diff
@@ -2692,6 +2692,48 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/transfers/{transferName}/validateTransfer": {
+      "post": {
+        "tags": [
+          "RecipientTransfers"
+        ],
+        "summary": "Validates if the products can be transfered in the context of the given transfer name.",
```
Comment (nschonni) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]:
```suggestion
        "summary": "Validates if the products can be transferred in the context of the given transfer name.",
```

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ValidateTransfer.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "properties": {
+              "productId": "subscriptionId",
+              "status": "Succeeded",
```
Comment (ryansbenson) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ValidateTransfer.json]: Is this a valid example? Success status with error result. Ideally examples represent real world interaction, not just an example of the schema.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 3]
```diff
@@ -2692,6 +2692,48 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/transfers/{transferName}/validateTransfer": {
```
[the remark this answers is not in the document]
Comment (jbnpaul) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: Removed the nextLink from the response.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ValidateTransfer.json (hunk 2 above)
Comment (jbnpaul) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/examples/ValidateTransfer.json]: Thanks. 
Fixed the example.

**Review by ryansbenson — APPROVED** [review]

**Review by tjprescott — APPROVED** [review]

**Pull request merged**